packages/chat-stateful-client/review/chat-stateful-client.api.md

```diff
@@ -21,14 +22,38 @@ export type ChatClientState = {
     threads: {
         [key: string]: ChatThreadClientState;
     };
+    latestErrors: ChatErrors;
 };
 
+// @public
+export class ChatError extends Error {
+    constructor(target: ChatErrorTargets, inner: Error);
+    inner: Error;
+    target: ChatErrorTargets;
+}
+
+// @public
+export type ChatErrors = {
+    [target in ChatErrorTargets]: Error;
+};
+
+// @public
+export type ChatErrorTargets = ChatObjectMethodNames<'ChatClient', ChatClient> | ChatObjectMethodNames<'ChatThreadClient', ChatThreadClient>;
+
 // @public (undocumented)
 export type ChatMessageWithStatus = ChatMessage & {
     clientMessageId?: string;
     status: MessageStatus;
 };
 
+// @public
+export type ChatMethodName<T, K extends keyof T> = T[K] extends Function ? (K extends string ? K : never) : never;
+
+// @public
+export type ChatObjectMethodNames<TName extends string, T> = {
```

Comment:
This is brilliant! Love this one.
Do we get a chance to validate this one for minimized js bundle(js-uglifier vs js minify)? I wonder if this still works, or it might be working in another unexpected way. Looks like this is generating the object name + function name in run time, curious what it would be like when code gets minified

Reply:
Right you are.

The type definition here is OK, that's anyway only for type safety, and doesn't survive in the minified code.

But the logic that tried to list the valid operations in `useAdaptedSelectors` was wrong. I dropped it. Please take another look.